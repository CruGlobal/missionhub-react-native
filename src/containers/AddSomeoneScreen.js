import React, { Component } from 'react';
import { connect } from 'react-redux';
import { translate } from 'react-i18next';

import IconMessageScreen from './IconMessageScreen';

@translate('addContact')
class AddSomeoneScreen extends Component {
  render() {
    const { t } = this.props;
    
    return (
      <IconMessageScreen
        mainText={t('message')}
        buttonText={t('addSomeone').toUpperCase()}
        nextScreen="SetupPerson"
        iconPath={require('../../assets/images/add_someone.png')}
      />
    );
  }
}

export default connect()(AddSomeoneScreen);
