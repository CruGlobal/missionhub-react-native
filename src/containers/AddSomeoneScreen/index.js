import React, {Component} from 'react';
import {connect} from 'react-redux';
import IconMessageScreen from '../IconMessageScreen/index';

class AddSomeoneScreen extends Component {
  constructor(props) {
    super(props);
  }

  render() {
    const message = 'Growing closer to God involves helping others experience Him. Who do you want to take steps of faith with?';
    return <IconMessageScreen mainText={message} buttonText="ADD SOMEONE" nextScreen="MainTabs"  iconPath={require('../../../assets/images/add_someone.png')} />;
  }
}

export default connect()(AddSomeoneScreen);
