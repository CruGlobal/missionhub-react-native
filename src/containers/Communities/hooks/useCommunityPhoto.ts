import { ImageSourcePropType } from 'react-native';

import { orgIsGlobal } from '../../../utils/common';
import GLOBAL_COMMUNITY_IMAGE from '../../../../assets/images/globalCommunityImage.png';

export const useCommunityPhoto = (
  communityId: string,
  communityPhotoUrl?: string | null,
): ImageSourcePropType => {
  if (communityPhotoUrl) {
    return { uri: communityPhotoUrl };
  } else if (orgIsGlobal({ id: communityId })) {
    return GLOBAL_COMMUNITY_IMAGE;
  } else {
    return {};
  }
};
