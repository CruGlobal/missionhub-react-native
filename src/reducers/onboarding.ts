import { LOGOUT, LogoutAction } from '../constants';
import {
  SET_ONBOARDING_PERSON_ID,
  SET_ONBOARDING_COMMUNITY,
  SKIP_ONBOARDING_ADD_PERSON,
  SetOnboardingPersonIdAction,
  SetOnboardingCommunityAction,
  SkipOnboardingAddPersonAction,
} from '../actions/onboarding';

export interface OnboardingState {
  personId: string;
  community: {
    id?: string;
    community_code: string;
    community_url: string;
  };
  skippedAddingPerson: boolean;
}

const initialState: OnboardingState = {
  personId: '',
  community: {
    id: undefined,
    community_code: '',
    community_url: '',
  },
  skippedAddingPerson: false,
};

type OnboardingActionTypes =
  | SetOnboardingPersonIdAction
  | SetOnboardingCommunityAction
  | SkipOnboardingAddPersonAction
  | LogoutAction;

export const onboardingReducer = (
  state: OnboardingState = initialState,
  action: OnboardingActionTypes,
) => {
  switch (action.type) {
    case SET_ONBOARDING_PERSON_ID:
      return { ...state, personId: action.personId };
    case SET_ONBOARDING_COMMUNITY:
      return {
        ...state,
        community: action.community,
      };
    case SKIP_ONBOARDING_ADD_PERSON:
      return {
        ...state,
        skippedAddingPerson: true,
      };
    case LOGOUT:
      return initialState;
    default:
      return state;
  }
};
