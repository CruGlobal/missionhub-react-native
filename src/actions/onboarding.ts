/* eslint-disable @typescript-eslint/no-explicit-any */

import uuidv4 from 'uuid/v4';
import { ThunkDispatch } from 'redux-thunk';
import { AnyAction } from 'redux';
import appsFlyer from 'react-native-appsflyer';

import { AuthState } from '../reducers/auth';
import { Person } from '../reducers/people';
import { OnboardingState } from '../reducers/onboarding';
import { OrganizationsState } from '../reducers/organizations';
import {
  ACTIONS,
  NOTIFICATION_PROMPT_TYPES,
  LOAD_PERSON_DETAILS,
} from '../constants';
import { rollbar } from '../utils/rollbar.config';
import { CELEBRATION_SCREEN } from '../containers/CelebrationScreen';
import { REQUESTS } from '../api/routes';
import { COMMUNITY_TABS } from '../containers/Communities/Community/constants';

import callApi from './api';
import { getMe } from './person';
import { navigatePush } from './navigation';
import { checkNotifications } from './notifications';
import { trackActionWithoutData } from './analytics';
import { joinCommunity } from './organizations';
import { updateLocaleAndTimezone } from './auth/userData';

export const START_ONBOARDING = 'START_ONBOARDING';
export const FINISH_ONBOARDING = 'FINISH_ONBOARDING';
export const SET_ONBOARDING_PERSON_ID = 'SET_ONBOARDING_PERSON_ID';
export const SET_ONBOARDING_COMMUNITY = 'SET_ONBOARDING_COMMUNITY_ID';
export const SKIP_ONBOARDING_ADD_PERSON = 'SKIP_ONBOARDING_ADD_PERSON';

export interface StartOnboardingAction {
  type: typeof START_ONBOARDING;
}

export interface FinishOnboardingAction {
  type: typeof FINISH_ONBOARDING;
}

export interface SetOnboardingPersonIdAction {
  type: typeof SET_ONBOARDING_PERSON_ID;
  personId: string;
}

export interface SetOnboardingCommunityAction {
  type: typeof SET_ONBOARDING_COMMUNITY;
  community: {
    id: string;
    community_code: string;
    community_url: string;
  };
}

export interface SkipOnboardingAddPersonAction {
  type: typeof SKIP_ONBOARDING_ADD_PERSON;
}

export const setOnboardingPersonId = (
  personId: string,
): SetOnboardingPersonIdAction => ({
  type: SET_ONBOARDING_PERSON_ID,
  personId,
});

export const setOnboardingCommunity = (community: {
  id: string;
  community_code: string;
  community_url: string;
}): SetOnboardingCommunityAction => ({
  type: SET_ONBOARDING_COMMUNITY,
  community,
});

export const skipOnboardingAddPerson = (): SkipOnboardingAddPersonAction => ({
  type: SKIP_ONBOARDING_ADD_PERSON,
});

export const startOnboarding = () => (
  dispatch: ThunkDispatch<{}, {}, AnyAction>,
) => {
  dispatch(trackActionWithoutData(ACTIONS.ONBOARDING_STARTED));
  dispatch({ type: START_ONBOARDING });
  appsFlyer.trackEvent(
    ACTIONS.ONBOARDING_STARTED.name,
    ACTIONS.ONBOARDING_STARTED,
  );
};

export function createMyPerson(firstName: string, lastName: string) {
  const data = {
    code: uuidv4(),
    first_name: firstName,
    last_name: lastName,
  };

  // eslint-disable-next-line @typescript-eslint/no-explicit-any
  return async (dispatch: ThunkDispatch<{}, {}, any>) => {
    await dispatch(callApi(REQUESTS.CREATE_MY_PERSON, {}, data));
    // @ts-ignore
    const me = ((await dispatch(getMe())) as unknown) as Person;
    dispatch(updateLocaleAndTimezone());

    rollbar.setPerson(me.id);

    dispatch({
      type: LOAD_PERSON_DETAILS,
      person: me,
    });

    return me;
  };
}

export const createPerson = (firstName: string, lastName: string) => async (
  // eslint-disable-next-line @typescript-eslint/no-explicit-any
  dispatch: ThunkDispatch<{}, {}, any>,
  getState: () => { auth: AuthState },
) => {
  const {
    person: { id: myId },
  } = getState().auth;
  const data = {
    data: {
      type: 'person',
      attributes: {
        first_name: firstName,
        last_name: lastName,
      },
    },
    included: [
      {
        type: 'contact_assignment',
        attributes: {
          assigned_to_id: myId,
        },
      },
    ],
  };

  const results = (await dispatch(
    callApi(REQUESTS.ADD_NEW_PERSON, {}, data),
    // eslint-disable-next-line @typescript-eslint/no-explicit-any
  )) as any;

  dispatch({
    type: LOAD_PERSON_DETAILS,
    person: results.response as Person,
  });

  return results;
};

export const skipAddPersonAndCompleteOnboarding = () => (
  dispatch: ThunkDispatch<{}, {}, AnyAction>,
) => {
  dispatch(skipOnboardingAddPerson());
  dispatch<any>(
    checkNotifications(NOTIFICATION_PROMPT_TYPES.ONBOARDING, () =>
      dispatch(navigatePush(CELEBRATION_SCREEN)),
    ),
  );
};

export const resetPersonAndCompleteOnboarding = () => (
  dispatch: ThunkDispatch<{}, {}, AnyAction>,
) => {
  dispatch(setOnboardingPersonId(''));
  dispatch<any>(
    checkNotifications(NOTIFICATION_PROMPT_TYPES.ONBOARDING, () =>
      dispatch(navigatePush(CELEBRATION_SCREEN)),
    ),
  );
};

export function joinStashedCommunity() {
  return async (
    dispatch: ThunkDispatch<{ auth: AuthState }, null, AnyAction>,
    getState: () => {
      onboarding: OnboardingState;
    },
  ) => {
    const {
      onboarding: { community },
    } = getState();

    community &&
      (await dispatch(
        joinCommunity(
          community.id,
          community.community_code,
          community.community_url,
        ),
      ));
  };
}

export function landOnStashedCommunityScreen() {
  return (
    dispatch: ThunkDispatch<
      { organizations: OrganizationsState },
      null,
      AnyAction
    >,
    getState: () => { onboarding: OnboardingState },
  ) => {
    dispatch(
      navigatePush(COMMUNITY_TABS, {
        communityId: getState().onboarding.community?.id,
      }),
    );
    dispatch(trackActionWithoutData(ACTIONS.SELECT_JOINED_COMMUNITY));
  };
}
