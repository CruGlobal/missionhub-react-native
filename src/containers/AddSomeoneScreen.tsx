import React from 'react';
import { connect } from 'react-redux';
import { useTranslation } from 'react-i18next';
import { ThunkDispatch, ThunkAction } from 'redux-thunk';

import { navigatePush } from '../actions/navigation';
import { disableBack } from '../utils/common';
import { skipOnboarding } from '../actions/onboardingProfile';
import { useDisableBack } from '../utils/hooks/useDisableBack';

import { SETUP_PERSON_SCREEN } from './SetupPersonScreen';
import IconMessageScreen from './IconMessageScreen';

interface AddSomeoneScreenProps {
  // eslint-disable-next-line @typescript-eslint/no-explicit-any
  dispatch: ThunkDispatch<any, null, never>;
  // eslint-disable-next-line @typescript-eslint/no-explicit-any
  next?: (props?: { skip: boolean }) => ThunkAction<void, any, null, never>; // TODO: make next required when only used in flows
}

const AddSomeoneScreen = ({ dispatch, next }: AddSomeoneScreenProps) => {
  const { t } = useTranslation('addContact');

  const enableBack = useDisableBack();

  const handleNavigate = (skip = false) => {
    enableBack();

    if (next) {
      return dispatch(next({ skip }));
    }

    dispatch(skip ? skipOnboarding() : navigatePush(SETUP_PERSON_SCREEN));
  };

  const skip = () => handleNavigate(true);

  return (
    <IconMessageScreen
      mainText={t('message')}
      onComplete={handleNavigate}
      buttonText={t('addSomeone')}
      iconPath={require('../../assets/images/add_someone.png')}
      onSkip={skip}
    />
  );
};

export default connect()(AddSomeoneScreen);
export const ADD_SOMEONE_SCREEN = 'nav/ADD_SOMEONE';
