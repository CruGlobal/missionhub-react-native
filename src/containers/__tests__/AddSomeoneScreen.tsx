import 'react-native';
import React from 'react';
import { fireEvent } from 'react-native-testing-library';

import AddSomeoneScreen from '../AddSomeoneScreen';
import { renderWithContext } from '../../../testUtils';
import { useLogoutOnBack } from '../../utils/hooks/useLogoutOnBack';
import { useAnalytics } from '../../utils/hooks/useAnalytics';
import { skipOnboarding } from '../../actions/onboarding';

jest.mock('../../actions/onboarding');
jest.mock('../../utils/hooks/useLogoutOnBack');
jest.mock('../../utils/hooks/useAnalytics');

const next = jest.fn();
const back = jest.fn();
const nextResult = { type: 'next' };
const skipOnboardingResult = { type: 'skip onboarding' };

beforeEach(() => {
  (next as jest.Mock).mockReturnValue(nextResult);
  (skipOnboarding as jest.Mock).mockReturnValue(skipOnboardingResult);
  (useLogoutOnBack as jest.Mock).mockReturnValue(back);
  (useAnalytics as jest.Mock).mockClear();
});

it('renders correctly', () => {
  renderWithContext(<AddSomeoneScreen next={next} />).snapshot();
});

it('renders without skip button correctly', () => {
  renderWithContext(
    <AddSomeoneScreen next={next} hideSkipBtn={true} />,
  ).snapshot();
});

it('renders without back button correctly', () => {
  (useLogoutOnBack as jest.Mock).mockReturnValue(null);

  renderWithContext(
    <AddSomeoneScreen next={next} enableBackButton={false} />,
  ).snapshot();
});

it('tracks screen change on mount', () => {
  renderWithContext(<AddSomeoneScreen next={next} />);

  expect(useAnalytics).toHaveBeenCalledWith(['onboarding', 'add someone']);
});

describe('onComplete', () => {
  it('calls next', () => {
    const { getByTestId, store } = renderWithContext(
      <AddSomeoneScreen next={next} />,
    );

    fireEvent.press(getByTestId('bottomButton'));

    expect(store.getActions()).toEqual([nextResult]);
    expect(next).toHaveBeenCalledWith({ skip: false });
  });
});

describe('onSkip prop', () => {
  it('calls next', () => {
    const { getByTestId, store } = renderWithContext(
      <AddSomeoneScreen next={next} />,
    );

    fireEvent.press(getByTestId('skipButton'));

    expect(store.getActions()).toEqual([nextResult]);
    expect(next).toHaveBeenCalledWith({ skip: true });
  });
});

describe('onBack prop', () => {
  describe('enableBackButton', () => {
    it('calls callback from useLogoutOnBack', () => {
      const { getByTestId } = renderWithContext(
        <AddSomeoneScreen next={next} />,
      );

      fireEvent.press(getByTestId('BackButton'));

      expect(useLogoutOnBack).toHaveBeenCalledWith(true, false);
      expect(back).toHaveBeenCalledWith();
    });
  });

  describe('logoutOnBack', () => {
    it('calls callback from useLogoutOnBack', () => {
      const { getByTestId } = renderWithContext(
        <AddSomeoneScreen next={next} logoutOnBack={true} />,
      );

      fireEvent.press(getByTestId('BackButton'));

      expect(useLogoutOnBack).toHaveBeenCalledWith(true, true);
      expect(back).toHaveBeenCalledWith();
    });
  });
});
