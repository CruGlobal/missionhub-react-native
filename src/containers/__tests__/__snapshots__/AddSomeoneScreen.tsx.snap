// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<View
  style={
    Object {
      "backgroundColor": "#007398",
      "flex": 1,
    }
  }
>
  <RCTSafeAreaView
    emulateUnlessSupported={true}
    style={
      Object {
        "backgroundColor": "transparent",
      }
    }
  >
    <View
      style={
        Array [
          Object {
            "flexDirection": "row",
            "height": 56,
            "paddingTop": 0,
          },
          undefined,
          null,
        ]
      }
    >
      <View
        style={
          Array [
            Object {
              "paddingLeft": 5,
            },
            Object {
              "alignItems": "flex-start",
              "flex": 1,
              "justifyContent": "center",
            },
          ]
        }
      >
        <View
          style={
            Array [
              Object {
                "alignItems": "flex-start",
                "alignSelf": "flex-start",
                "justifyContent": "center",
              },
              undefined,
            ]
          }
        >
          <View
            accessible={true}
            focusable={true}
            onClick={[Function]}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              Object {
                "opacity": 1,
              }
            }
            testID="IconButton"
          >
            <View
              style={
                Array [
                  Object {
                    "backgroundColor": "transparent",
                  },
                  null,
                  Object {},
                  null,
                ]
              }
            >
              <Text
                allowFontScaling={false}
                style={
                  Array [
                    Object {
                      "color": undefined,
                      "fontSize": 12,
                    },
                    Array [
                      Object {
                        "color": "#ffffff",
                      },
                      Object {
                        "fontSize": 18,
                      },
                      Array [
                        Object {
                          "margin": 10,
                        },
                        undefined,
                      ],
                    ],
                    Object {
                      "fontFamily": "icomoon",
                      "fontStyle": "normal",
                      "fontWeight": "normal",
                    },
                    Object {},
                  ]
                }
              >
                î¤ƒ
              </Text>
            </View>
          </View>
        </View>
      </View>
      <View
        style={
          Array [
            Object {
              "paddingRight": 5,
            },
            Object {
              "alignItems": "flex-end",
              "flex": 1,
              "justifyContent": "center",
            },
          ]
        }
      >
        <View
          accessible={true}
          focusable={true}
          onClick={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Object {
              "opacity": 1,
            }
          }
          testID="skipButton"
        >
          <View
            style={
              Array [
                Object {
                  "backgroundColor": "transparent",
                },
                null,
                Array [
                  Object {
                    "padding": 15,
                  },
                  undefined,
                ],
                null,
              ]
            }
          >
            <Text
              style={
                Array [
                  Object {
                    "color": "#ffffff",
                    "fontFamily": "SourceSansPro-Regular",
                    "fontSize": 14,
                    "fontStyle": "normal",
                    "fontWeight": "bold",
                    "letterSpacing": 1,
                    "lineHeight": 18,
                    "textAlign": "center",
                    "textAlignVertical": "center",
                    "textTransform": "uppercase",
                  },
                  Array [
                    Object {
                      "color": "#ffffff",
                      "fontFamily": "SourceSansPro-Regular",
                      "fontSize": 14,
                      "fontStyle": "normal",
                      "fontWeight": "normal",
                      "lineHeight": 20,
                      "textAlignVertical": "center",
                      "textTransform": "none",
                    },
                    undefined,
                  ],
                ]
              }
            >
              SKIP
            </Text>
          </View>
        </View>
      </View>
    </View>
  </RCTSafeAreaView>
  <View
    style={
      Array [
        Object {
          "alignItems": "center",
          "flex": 1,
          "justifyContent": "center",
          "paddingHorizontal": 60,
        },
        Object {
          "alignItems": "center",
          "flex": 1,
          "justifyContent": "center",
        },
      ]
    }
  >
    <View
      style={
        Array [
          Object {},
          Object {
            "alignItems": "flex-start",
            "flex": 4,
            "justifyContent": "center",
          },
        ]
      }
    >
      <Image
        source={
          Object {
            "testUri": "../../../assets/images/add_someone.png",
          }
        }
        style={
          Object {
            "height": 80,
            "resizeMode": "contain",
            "width": 80,
          }
        }
      />
      <Text
        style={
          Object {
            "color": "#ffffff",
            "fontFamily": "SourceSansPro-Regular",
            "fontSize": 24,
            "fontStyle": "normal",
            "fontWeight": "normal",
            "lineHeight": 32,
            "paddingVertical": 10,
            "textAlign": "left",
            "textAlignVertical": "center",
            "textTransform": "none",
          }
        }
      >
        Growing closer to God involves helping others experience Him.

Take a moment and pray. Who do you want to take steps of faith with?
      </Text>
    </View>
    <RCTSafeAreaView
      emulateUnlessSupported={true}
      style={
        Object {
          "bottom": 20,
          "left": 50,
          "position": "absolute",
        }
      }
    >
      <View
        accessible={true}
        focusable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "opacity": 1,
          }
        }
        testID="bottomButton"
      >
        <View
          style={
            Array [
              Object {
                "alignItems": "center",
                "backgroundColor": "#3CC8E6",
                "height": 60,
                "justifyContent": "center",
              },
              null,
              Array [
                Object {
                  "alignItems": "center",
                  "height": 48,
                  "width": 650,
                },
                undefined,
              ],
              Object {
                "borderRadius": 50,
              },
            ]
          }
        >
          <Text
            style={
              Array [
                Object {
                  "color": "#ffffff",
                  "fontFamily": "SourceSansPro-Regular",
                  "fontSize": 14,
                  "fontStyle": "normal",
                  "fontWeight": "bold",
                  "letterSpacing": 1,
                  "lineHeight": 18,
                  "textAlign": "center",
                  "textAlignVertical": "center",
                  "textTransform": "uppercase",
                },
                Array [
                  undefined,
                ],
              ]
            }
          >
            ADD SOMEONE
          </Text>
        </View>
      </View>
    </RCTSafeAreaView>
  </View>
</View>
`;

exports[`renders without back button correctly 1`] = `
<View
  style={
    Object {
      "backgroundColor": "#007398",
      "flex": 1,
    }
  }
>
  <RCTSafeAreaView
    emulateUnlessSupported={true}
    style={
      Object {
        "backgroundColor": "transparent",
      }
    }
  >
    <View
      style={
        Array [
          Object {
            "flexDirection": "row",
            "height": 56,
            "paddingTop": 0,
          },
          undefined,
          null,
        ]
      }
    >
      <View
        style={
          Array [
            Object {
              "paddingLeft": 5,
            },
            Object {
              "alignItems": "flex-start",
              "flex": 1,
              "justifyContent": "center",
            },
          ]
        }
      />
      <View
        style={
          Array [
            Object {
              "paddingRight": 5,
            },
            Object {
              "alignItems": "flex-end",
              "flex": 1,
              "justifyContent": "center",
            },
          ]
        }
      >
        <View
          accessible={true}
          focusable={true}
          onClick={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Object {
              "opacity": 1,
            }
          }
          testID="skipButton"
        >
          <View
            style={
              Array [
                Object {
                  "backgroundColor": "transparent",
                },
                null,
                Array [
                  Object {
                    "padding": 15,
                  },
                  undefined,
                ],
                null,
              ]
            }
          >
            <Text
              style={
                Array [
                  Object {
                    "color": "#ffffff",
                    "fontFamily": "SourceSansPro-Regular",
                    "fontSize": 14,
                    "fontStyle": "normal",
                    "fontWeight": "bold",
                    "letterSpacing": 1,
                    "lineHeight": 18,
                    "textAlign": "center",
                    "textAlignVertical": "center",
                    "textTransform": "uppercase",
                  },
                  Array [
                    Object {
                      "color": "#ffffff",
                      "fontFamily": "SourceSansPro-Regular",
                      "fontSize": 14,
                      "fontStyle": "normal",
                      "fontWeight": "normal",
                      "lineHeight": 20,
                      "textAlignVertical": "center",
                      "textTransform": "none",
                    },
                    undefined,
                  ],
                ]
              }
            >
              SKIP
            </Text>
          </View>
        </View>
      </View>
    </View>
  </RCTSafeAreaView>
  <View
    style={
      Array [
        Object {
          "alignItems": "center",
          "flex": 1,
          "justifyContent": "center",
          "paddingHorizontal": 60,
        },
        Object {
          "alignItems": "center",
          "flex": 1,
          "justifyContent": "center",
        },
      ]
    }
  >
    <View
      style={
        Array [
          Object {},
          Object {
            "alignItems": "flex-start",
            "flex": 4,
            "justifyContent": "center",
          },
        ]
      }
    >
      <Image
        source={
          Object {
            "testUri": "../../../assets/images/add_someone.png",
          }
        }
        style={
          Object {
            "height": 80,
            "resizeMode": "contain",
            "width": 80,
          }
        }
      />
      <Text
        style={
          Object {
            "color": "#ffffff",
            "fontFamily": "SourceSansPro-Regular",
            "fontSize": 24,
            "fontStyle": "normal",
            "fontWeight": "normal",
            "lineHeight": 32,
            "paddingVertical": 10,
            "textAlign": "left",
            "textAlignVertical": "center",
            "textTransform": "none",
          }
        }
      >
        Growing closer to God involves helping others experience Him.

Take a moment and pray. Who do you want to take steps of faith with?
      </Text>
    </View>
    <RCTSafeAreaView
      emulateUnlessSupported={true}
      style={
        Object {
          "bottom": 20,
          "left": 50,
          "position": "absolute",
        }
      }
    >
      <View
        accessible={true}
        focusable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "opacity": 1,
          }
        }
        testID="bottomButton"
      >
        <View
          style={
            Array [
              Object {
                "alignItems": "center",
                "backgroundColor": "#3CC8E6",
                "height": 60,
                "justifyContent": "center",
              },
              null,
              Array [
                Object {
                  "alignItems": "center",
                  "height": 48,
                  "width": 650,
                },
                undefined,
              ],
              Object {
                "borderRadius": 50,
              },
            ]
          }
        >
          <Text
            style={
              Array [
                Object {
                  "color": "#ffffff",
                  "fontFamily": "SourceSansPro-Regular",
                  "fontSize": 14,
                  "fontStyle": "normal",
                  "fontWeight": "bold",
                  "letterSpacing": 1,
                  "lineHeight": 18,
                  "textAlign": "center",
                  "textAlignVertical": "center",
                  "textTransform": "uppercase",
                },
                Array [
                  undefined,
                ],
              ]
            }
          >
            ADD SOMEONE
          </Text>
        </View>
      </View>
    </RCTSafeAreaView>
  </View>
</View>
`;

exports[`renders without skip button correctly 1`] = `
<View
  style={
    Object {
      "backgroundColor": "#007398",
      "flex": 1,
    }
  }
>
  <RCTSafeAreaView
    emulateUnlessSupported={true}
    style={
      Object {
        "backgroundColor": "transparent",
      }
    }
  >
    <View
      style={
        Array [
          Object {
            "flexDirection": "row",
            "height": 56,
            "paddingTop": 0,
          },
          undefined,
          null,
        ]
      }
    >
      <View
        style={
          Array [
            Object {
              "paddingLeft": 5,
            },
            Object {
              "alignItems": "flex-start",
              "flex": 1,
              "justifyContent": "center",
            },
          ]
        }
      >
        <View
          style={
            Array [
              Object {
                "alignItems": "flex-start",
                "alignSelf": "flex-start",
                "justifyContent": "center",
              },
              undefined,
            ]
          }
        >
          <View
            accessible={true}
            focusable={true}
            onClick={[Function]}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              Object {
                "opacity": 1,
              }
            }
            testID="IconButton"
          >
            <View
              style={
                Array [
                  Object {
                    "backgroundColor": "transparent",
                  },
                  null,
                  Object {},
                  null,
                ]
              }
            >
              <Text
                allowFontScaling={false}
                style={
                  Array [
                    Object {
                      "color": undefined,
                      "fontSize": 12,
                    },
                    Array [
                      Object {
                        "color": "#ffffff",
                      },
                      Object {
                        "fontSize": 18,
                      },
                      Array [
                        Object {
                          "margin": 10,
                        },
                        undefined,
                      ],
                    ],
                    Object {
                      "fontFamily": "icomoon",
                      "fontStyle": "normal",
                      "fontWeight": "normal",
                    },
                    Object {},
                  ]
                }
              >
                î¤ƒ
              </Text>
            </View>
          </View>
        </View>
      </View>
      <View
        style={
          Array [
            Object {
              "paddingRight": 5,
            },
            Object {
              "alignItems": "flex-end",
              "flex": 1,
              "justifyContent": "center",
            },
          ]
        }
      />
    </View>
  </RCTSafeAreaView>
  <View
    style={
      Array [
        Object {
          "alignItems": "center",
          "flex": 1,
          "justifyContent": "center",
          "paddingHorizontal": 60,
        },
        Object {
          "alignItems": "center",
          "flex": 1,
          "justifyContent": "center",
        },
      ]
    }
  >
    <View
      style={
        Array [
          Object {},
          Object {
            "alignItems": "flex-start",
            "flex": 4,
            "justifyContent": "center",
          },
        ]
      }
    >
      <Image
        source={
          Object {
            "testUri": "../../../assets/images/add_someone.png",
          }
        }
        style={
          Object {
            "height": 80,
            "resizeMode": "contain",
            "width": 80,
          }
        }
      />
      <Text
        style={
          Object {
            "color": "#ffffff",
            "fontFamily": "SourceSansPro-Regular",
            "fontSize": 24,
            "fontStyle": "normal",
            "fontWeight": "normal",
            "lineHeight": 32,
            "paddingVertical": 10,
            "textAlign": "left",
            "textAlignVertical": "center",
            "textTransform": "none",
          }
        }
      >
        Growing closer to God involves helping others experience Him.

Take a moment and pray. Who do you want to take steps of faith with?
      </Text>
    </View>
    <RCTSafeAreaView
      emulateUnlessSupported={true}
      style={
        Object {
          "bottom": 20,
          "left": 50,
          "position": "absolute",
        }
      }
    >
      <View
        accessible={true}
        focusable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "opacity": 1,
          }
        }
        testID="bottomButton"
      >
        <View
          style={
            Array [
              Object {
                "alignItems": "center",
                "backgroundColor": "#3CC8E6",
                "height": 60,
                "justifyContent": "center",
              },
              null,
              Array [
                Object {
                  "alignItems": "center",
                  "height": 48,
                  "width": 650,
                },
                undefined,
              ],
              Object {
                "borderRadius": 50,
              },
            ]
          }
        >
          <Text
            style={
              Array [
                Object {
                  "color": "#ffffff",
                  "fontFamily": "SourceSansPro-Regular",
                  "fontSize": 14,
                  "fontStyle": "normal",
                  "fontWeight": "bold",
                  "letterSpacing": 1,
                  "lineHeight": 18,
                  "textAlign": "center",
                  "textAlignVertical": "center",
                  "textTransform": "uppercase",
                },
                Array [
                  undefined,
                ],
              ]
            }
          >
            ADD SOMEONE
          </Text>
        </View>
      </View>
    </RCTSafeAreaView>
  </View>
</View>
`;
